src/main/java/bio/terra/service/policy/PolicyService.java

```diff
@@ -0,0 +1,139 @@
+package bio.terra.service.policy;
+
+import bio.terra.app.configuration.PolicyServiceConfiguration;
+import bio.terra.common.ExceptionUtils;
+import bio.terra.model.RepositoryStatusModelSystems;
+import bio.terra.policy.api.PublicApi;
+import bio.terra.policy.api.TpsApi;
+import bio.terra.policy.client.ApiClient;
+import bio.terra.policy.client.ApiException;
+import bio.terra.policy.model.TpsComponent;
+import bio.terra.policy.model.TpsObjectType;
+import bio.terra.policy.model.TpsPaoCreateRequest;
+import bio.terra.policy.model.TpsPolicyInputs;
+import bio.terra.service.policy.exception.PolicyConflictException;
+import bio.terra.service.policy.exception.PolicyServiceAPIException;
+import bio.terra.service.policy.exception.PolicyServiceAuthorizationException;
+import bio.terra.service.policy.exception.PolicyServiceDuplicateException;
+import bio.terra.service.policy.exception.PolicyServiceNotFoundException;
+import com.google.common.annotations.VisibleForTesting;
+import java.io.IOException;
+import java.util.UUID;
+import javax.annotation.Nullable;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+
+@Component
+public class PolicyService {
+  private static final Logger logger = LoggerFactory.getLogger(PolicyService.class);
+  private final PolicyServiceConfiguration policyServiceConfiguration;
+
+  @Autowired
+  public PolicyService(PolicyServiceConfiguration policyServiceConfiguration) {
+    this.policyServiceConfiguration = policyServiceConfiguration;
+    logger.info("TPS base path: '{}'", policyServiceConfiguration.getBasePath());
+  }
+
+  // -- Policy Attribute Object Interface --
+  public void createSnapshotPao(UUID snapshotId, @Nullable TpsPolicyInputs policyInputs) {
+    createPao(snapshotId, TpsObjectType.SNAPSHOT, policyInputs);
+  }
+
+  public void createPao(
+      UUID resourceId, TpsObjectType resourceType, @Nullable TpsPolicyInputs policyInputs) {
+    policyServiceConfiguration.tpsEnabledCheck();
```

Comment:
I'm curious about this "feature flag" -- what's the long-term intention?  Do we want environment-specific enforcement on whether or not these methods are called?  Ultimately would we want to remove this feature flag code when it's enabled everywhere?

Reply:
TPS is only available in dev right now, so this allows us to enable this per environment! We did something similar with RBS (although I think we never removed the enabled flag).

Reply:
Nice!

I’ve found that feature flag code can linger after you need it. What do you think about filing a preemptive ticket in the epic for its clean up, as well as updating the TPS hosts by environment? Awhile back we had a prod incident driven by me forgetting to update the ECM hosts by environment, and TDR prod mistakenly communicating with ECM dev.

Reply:
Good call! I created a ticket here: https://broadworkbench.atlassian.net/browse/DR-3029 and will update that with links to the code once everything is merged in.